src/types/mod.rs

```diff
@@ -2650,3 +2650,143 @@ mod transaction_fee_distribution {
+#[derive(Debug)]
+pub struct NetworkInfo {
+    pub node_id:             String,
+    pub peer_total_sent:     u64,
+    pub peer_total_received: u64,
+    pub avg_bps_in:          u64,
+    pub avg_bps_out:         u64,
+}
+
+/// A node can either be a 'Bootstrapper' or 'Node'.
+/// The 'Bootstrapper' is a special kind of node that does not participate
+/// in consensus but merely relay peers.
+/// The 'Node' is a node that participates in consensus either passively (only
+/// processing incoming blocks) or actively i.e. by being a baker.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Debug, Display, From, TryInto)]
+pub enum NodeType {
+    Bootstrapper,
+    Node,
+}
+
+impl TryFrom<i32> for NodeType {
+    type Error = anyhow::Error;
+
+    // Parse the node type.
+    // The node type is specified via an enum:
+    // * 0 => The node is a bootstrapper and is simply relaying peers.
+    // * 1 => The node is participating in consensus either actively (baking) or
+    //   passively (solely processing blocks).
+    // If the result is none of the above we return an error "Malformed node type."
+    fn try_from(v: i32) -> Result<Self, Self::Error> {
+        match v {
+            x if x == NodeType::Bootstrapper as i32 => Ok(NodeType::Bootstrapper),
+            x if x == NodeType::Node as i32 => Ok(NodeType::Node),
+            _ => Err(anyhow::anyhow!("Malformed node type")),
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Consensus related information for a node.
+pub enum ConsensusInfo {
+    // The consensus protocol is not running on the node.
+    // This only occurs when the node does not support the protocol on the chain or the node is a
+    // 'Bootstrapper'.
+    ConsensusNotRunning,
+    // The node is a passive member of the consensus. This means:
+    // * The node is processing blocks.
+    // * The node is relaying transactions and blocks onto the network.
+    // * The node is responding to catch up messages from its peers.
+    // * In particular this means that the node is __not__ baking blocks.
+    ConsensusPassive,
+    // The node has been configured with baker keys however it is not currently baking and
+    // possilby never will.
+    NotInCommittee,
+    // The baker keys are registered however the baker is not in the committee
+    // for the current 'Epoch'.
+    AddedButNotActiveInCommittee,
+    // The node has been configured with baker keys that does not match the account.
+    AddedButWrongKeys,
+    // The node is member of the baking committee.
+    Baker(crate::types::BakerId),
+    // The node is member of the baking and finalization committee.
+    Finalizer(crate::types::BakerId),
+}
+
+#[derive(Debug)]
+/// The status of the requested node.
+pub struct NodeInfo {
+    // The version of the node.
+    // Note. if the returned version from the node does
+    // not conform to semantic versioning then the parsing will fail.
```

Comment:
Why comment on that here? The reader of this documentation cannot really do much with this information.